ZCRMiOS/Classes/ZCRMDashBoardComponent.swift

```diff
@@ -0,0 +1,635 @@
+        DASHBOARD COMPONENT PROPERTIES ---------------X
+        Name: \(name)
+        Category: \(category.rawValue)
+        Type: \(componentType)
+        ReportID: \(String(describing: reportId))
+        Objective: \(objective?.rawValue ?? "nil")
+        Maximum Rows: \(String(describing: maximumRows))
+        Component Markers: \(String(describing: componentMarker))
+        VISUALIZATION PROPERTIES --------------------X
+        ColorPaletteName: \(colorPaletteName?.rawValue ?? "nil")
+        ColorPalette Starting Index: \(String(describing: colorPaletteStartingIndex))
+        
+        Segment Ranges: \(String(describing: segmentRanges))
+        
+        LastFetchedTime Label: \(String(describing: lastFetchedTimeLabel))
+        LastFetchedTime Value: \(String(describing: lastFetchedTimeValue))
+        <-------------------------------------------->
+        """
+    }
+}
+
+//MARK:- Getters
+extension ZCRMDashBoardComponent
+{
+    public func getComponentCategory() -> ComponentCategory
+    {
+        return category
+    }
+    
+    public func getComponentName() -> String
+    {
+        return name
+    }
+    
+    public func getLastFetchedTimeLabel() -> String?
+    {
+        return lastFetchedTimeLabel
+    }
+    
+    public func getLastFetchedTimeValue() -> String?
+    {
+        return lastFetchedTimeValue
+    }
+    
+    public func getReportID() -> Int64?
+    {
+        return reportId
+    }
+    
+    public func getMaximumRows() -> Int?
+    {
+        return maximumRows
+    }
+    
+    public func getObjective() -> Objective?
+    {
+        return objective
+    }
+    
+    public func getComponentType() -> String
+    {
+        return componentType
+    }
+    
+    public func getColorPaletteName() -> colorPalette?
+    {
+        return colorPaletteName
+    }
+    
+    public func getColorPaletteStartingIndex() -> Int?
+    {
+        return colorPaletteStartingIndex
+    }
+    
+    public func getSegmentRanges() -> [SegmentRanges]?
+    {
+        return segmentRanges
+    }
+    
+    public func getComponentMarkers() -> [ComponentMarkers]?
+    {
+        return componentMarker
+    }
+    
+    public func getComponentChunks() -> [ComponentChunks]
+    {
+        return componentChunks
+    }
+}
+
+//MARK:- Setters
+extension ZCRMDashBoardComponent
+{
+    public func setComponent(category: ComponentCategory?)
+    {
+        self.category = category ?? ComponentCategory.chart
+    }
+    
+    public func setComponent(name: String?)
+    {
+        self.name = name ?? String()
+    }
+    
+    public func setComponent(markers: [ComponentMarkers]?)
+    {
+        componentMarker = markers
+    }
+    
+    public func setLastFetchedTime(label: String?)
+    {
+        lastFetchedTimeLabel = label ?? String()
+    }
+    
+    public func setLastFetchedTime(value: String?)
+    {
+        lastFetchedTimeValue = value ?? String()
+    }
+    
+    public func setReport(id: Int64?)
```

Comment:
setReportId. 
If you say setReport, the argument should be a Report object type and not an id.